### DISCUSSION TITLE
Atrim 800 USB Controller Stuck at 7.17.2

### DISCUSSION THREAD

**Discussion by ioguy** [zwave-js/zwave-js#6816] [Request Support / Investigate Issue]
Just received a new Atrim 800 USB Controller and it came with firmware 7.17.2.

Using Z-Wave JS 12.5.6 to flash on a separate computer under docker.

I'm trying to update but having no success (in PC Controller or Z-Wave JS).  I tried 7.18.0, 7.18.2, 7.18.8 and 7.21.2.  I'm using BRD4204D which is what Atrim specifies on their website:
https://atrim.co/help/stick/atrim-stick-update-firmware-home-assistant-800-series.html

All give the following failure:

> 16:58:21.896 CNTRLR   Interview completed
> 16:58:21.897 DRIVER   No network key for S0 configured, communication with secure (S0) devices won't work!
> 16:58:21.897 DRIVER   No network key for S2 configured, communication with secure (S2) devices won't work!
> 16:58:21.897 DRIVER   No network key for Z-Wave Long Range configured, communication won't work!
> 16:58:21.898 DRIVER   driver ready
> Flashing firmware...
> 16:58:21.900 CNTRLR   Entering bootloader...
> 16:58:21.901 CNTRLR   Performing soft reset...
> 16:58:21.918 DRIVER   all queues busy
> 16:58:21.919 SERIAL » 0x01030008f4                                                         (5 bytes)
> 16:58:21.920 DRIVER » [REQ] [SoftReset]
> 16:58:21.924 SERIAL « [ACK]                                                                   (0x06)
> 16:58:21.925 CNTRLR   Waiting for the controller to reconnect...
> 16:58:21.925 DRIVER   all queues idle
> 16:58:23.159 SERIAL « 0x011a000a0300030201105e228570567a72735586595a6c74989f0196          (28 bytes)
> 16:58:23.160 SERIAL » [ACK]                                                                   (0x06)
> 16:58:23.161 DRIVER « [REQ] [SerialAPIStarted]
>                         wake up reason:        WatchdogReset
>                         watchdog enabled:      false
>                         generic device class:  0x02
>                         specific device class: 0x01
>                         always listening:      false
>                         supports Long Range:   true
> 16:58:23.162 CNTRLR   reconnected and restarted
> 16:58:23.162 SERIAL » 0x01030027db                                                         (5 bytes)
> 16:58:23.202 SERIAL « �
>                       Gecko Bootloader v2.00.01
>                       1. upload gbl
>                       2. run
>                       3. ebl info
>                       BL >
> 16:58:23.205 CNTRLR   [BOOTLOADER] version 2.00.01
> 16:58:23.206 CNTRLR   Entered bootloader
> 16:58:23.206 CNTRLR   Beginning firmware upload
> 16:58:23.207 SERIAL » 0x31                                                                 (1 bytes)
> 16:58:23.213 SERIAL « begin upload
> 16:58:23.213 CNTRLR   [BOOTLOADER] begin upload
> 16:58:23.217 SERIAL « 0x43                                                                 (1 bytes)
> 16:58:23.218 CNTRLR   [BOOTLOADER] awaiting input...
> 16:58:23.220 SERIAL » 0x0101feeb17a603080000000000000301010000fa0606fa10000000d0070200bc (133 bytes)
>                       b33256a225132ce9ead8d9f90707f924000000f80e0b4a76a2b63d99b3c9433122dda8b515d27e
>                       09631bf8b53e4598f0dc3a2012bcd603f90707f9ac07020015fdc254802c7652607ee8e20b781e
>                       317cf5329d7edc55b9a49f2c7a151099aa82529bfbb79d
> 16:58:23.236 SERIAL « 0x18                                                                 (1 bytes)
> 16:58:23.237 SERIAL « 0x18                                                                 (1 bytes)
> 16:58:23.238 SERIAL « 0x18                                                                 (1 bytes)
> 16:58:23.250 SERIAL « Serial upload aborted
> 16:58:23.250 CNTRLR   [BOOTLOADER] Serial upload aborted
> 16:58:23.251 SERIAL « file error 0x50
> 16:58:23.251 CNTRLR   [BOOTLOADER] file error 0x50
> 16:58:23.256 SERIAL « Gecko Bootloader v2.00.01
>                       1. upload gbl
>                       2. run
>                       3. ebl info
>                       BL >
> 16:58:23.257 CNTRLR   OTW update was aborted by the bootloader. file error 0x50
> Firmware update failed: Error_Aborted
> 16:58:23.258 CNTRLR   Leaving bootloader...

The SiLabs release notes does mention some info on upgrading from 7.17 but I'm not sure what to do (or if it relates):

> Be aware that 800 products based on SDK v7.17.x do not support upgrade of Secure Element firmware over the air (OTA). However, a migration path exists to upgrade both main bootloader and Secure Element firmware to enable support of this feature. See INS14895: Instruction for How to Use Tiny App regarding the upgrade path. The 800-based SDK v7.18.x supports upgrade of Secure Element firmware over the air (OTA).
> The 8 kB reduction of the Z-Wave protocol NVM3 file system has an impact when making OTA firmware update on 800-based ap-plications deployed on version 7.17.2 and earlier. To make an OTA firmware update from 7.17.2 to 7.18.1/2 requires that 7.18.1/2 is modified to keep the same NVM3 protocol size as 7.17.2. This can be configured by the define NVM3_DEFAULT_NVM_SIZE when building 7.18.1/2.

Any ideas on how to get started or if I should just return it?

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment]
File error 0x50 means the bootloader considers the update incompatible. How does the Atrim stick show up in Z-Wave JS UI? If they have their own IDS, you probably need their firmware and not the pre-built ones from Silabs.
Although they are saying to use the Silabs firmware...

Have you asked Atrim support?

**Reply by ioguy** [reply]
Shows up as:
<img width="815" alt="image" src="https://github.com/zwave-js/node-zwave-js/assets/15815469/85cae654-66cf-48d9-8867-77dfc5156ef5">

I've emailed Atrim 3 times - radio silence.  Also can't find anything online about it.

I'd just pickup a ZST39 but I don't want to be locked at Zooz's firmware levels.

**Reply by AlCalzone** [reply]
> Shows up as: [...]

Based on that, the default firmwares should work. No idea why they don't.

**Reply by ioguy** [reply]
I just received a reply from Atrim (maybe they read these forums).  Apparently it's an SiLabs bug where the memory allocation/partitioning is different 7.17 and it cannot be upgraded.  7.21.X may eventually have a fix for this.  They pointed me to this discussion but there's not a lot of info, nor does an SiLabs employee confirm anything:
https://community.silabs.com/s/question/0D58Y0000Ae0lcMSQQ/update-firmware-from-1772-to-17201-fails-with-file-error-0x18?language=en_US

Guess I'll just sit and wait.

**Reply by AlCalzone** [reply]
I've seen that issue, but AFAIK it was specifically for upgrading to 7.20.0/1 and has been fixed since. Sit and wait it is then.

**Comment by Hedda** [comment]
FYI, while I understand that it that should be no problem flashing the Silicon Labs Z-Wave 700 series or 800 series controller hardware, be warned that there is currently also an unrelated serious bug in all 7.19.x to 7.21.2 versions of Z-Wave Controller firmware (i.e. all firmware released after 7.18.8) that causes the Z-Wave Controller to randomly hang during transmission until it is restarted. It is currently unclear if this bug is fixed in a later firmware version." though Silabs have said they that they are aware and plan to fix it. So for now do not upgrade to anything higher than 7.18.8 firmware. As such it is currently not recommended to upgrade to anything higher than v7.18.8 is recommended today. See:

https://community.silabs.com/s/question/0D5Vm0000036y3PKAQ/controller-reports-being-jammed-during-usage-and-provokes-the-zwave-network-to-stall-for-several-seconds

"_During the validation of the fix attempt we identified that the solution is not complete and with the refactoring efforts needed, it was not possible to integrate it into the 7.21.2 release. The team is well aware of the impact of this issue and is actively working on it with the highest priority. We will provide an update with the new ETA when confirmed by our R&D._"

You should also be aware there are additional known issues or bugs that can sometimes cause both TX Power Level + Measured Power Levels and/or RF Region settings to reset themselves when flashing Z-Wave Controller firmware upgrade, meaning that you sometimes manually have to re-configure those to correct settings after upgrade firmware, (the general recommendation there seems to be to upgrade one version at a time and not jump over a version). There are more related discussions here:

* https://github.com/zwave-js/node-zwave-js/discussions
  * https://github.com/zwave-js/node-zwave-js/discussions/2544
  * https://github.com/zwave-js/node-zwave-js/discussions/2549
  * https://github.com/zwave-js/node-zwave-js/discussions/4112
  * https://github.com/zwave-js/node-zwave-js/discussions/5615

PS: Regardless of which Z-Wave Controller you are using, because Z-Wave radios use use low-power and UHF (Ultra High Frequency) to only send short packets which generally means relativly poor [radio propagation](https://en.wikipedia.org/wiki/Radio_propagation) though many types of building materials, the general best practice recommendation is to take preventive actions to avoid keeping sources of [EMI/RMI/EMF-interference](https://en.wikipedia.org/wiki/Electromagnetic_interference) too close to the Z-Wave Controller adapter, and the easiest way to do so if you have a USB radio adapter/dongle is to connect it via long shielded USB extension cable to a USB 2.0 port or USB 2.0 hub to be able to get it a bit away on its own (as well as especially making sure to keep it away from USB 3.x/4.x  ports/devices/peripherals/cables as such are infamous for causing EMI/RMI/EMF-interference that affect RF IoT devices. Of course, since Z-Wave network's range/coverage depends heavily on [mesh networking](https://en.wikipedia.org/wiki/Mesh_networking) by design you also want to make sure to add many always-on/always-connected/always-powered mains-powered Z-Wave devices at strategic locations in your home that act as Z-Wave Repeater devices (sometimes also referred to as Z-Wave Range Extenders or Z-Wave Routers).

**Reply by ioguy** [reply]
I'm not concerned about the failed to transmit bug.  My current system is on 7.21.3 and I get it locking up all the time - it hiccups and starts up again (I've never had to restart it myself).

I want to switch to the 800 for the additional speed/range but unfortunately I can't NVM restore onto 7.17.2.
